core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -6137,39 +6208,6 @@ private Function toLimitFunction(
         return func;
     }
 
-    // UNION_CAST_MATRIX captures all the combinations of "left" and "right" column types
```

Comment:
@jerrinot union cast matrix is generate by thus code, we should not be manually updating it and the code must remain in place commented out

Reply:
@bluestreak01 sees tests in `SqlCodeGeneratorTest`. This is a change from a @amunra's commit: Not only is the generation kept in place (and printed), but Adam also added a test to check that the stored matrix is the same as the one auto-generated at test-time.

Reply:
TL;DR: The generation code and the matrix had _already_ diverged, I brought them back together and added tests to they don't diverge in the future.

Steps I took:
* I uncommented the code and marked it as `@TestOnly`.
* Then added a test to assert it's the same as the table that's there.
* It diverged significantly.
* I kept whatever was checked in the table and updated the generation code.
* The generation code now outputs a new table during tests should we add new types.

I did this in the `nd_arr` branch first (as part of test case failures fixes), then as I was asking about code in this area Jaromir spotted a bug and fixed it.